File: samples/cs/sqrt/sqrt.cs
using System;

namespace SquareRoot {
    class Exec {

        static double FindSqrt(double number) {
            double x = number;
            double y = 1.0;
            const double e = 0.0000001;
            while ((x - y) > e) {
                x = (x + y) / 2;
                y = number / x;
            }
            return x;
        }

        static void Main(string[] args) {
            if (args.Length != 1) {
                Console.Error.WriteLine("Error: Expected to receive 1 argument ./sqrt <number>.");
                System.Environment.Exit(1);
            }
            double number = 0;
            try {
                number = Double.Parse(args[0]);
            } catch (System.FormatException) {
                Console.Error.WriteLine(string.Format("Error: Expected to receive 1 argument that was a number. Received {0}", args[0]));
                System.Environment.Exit(2);
            }
            Console.WriteLine(FindSqrt(number));
        }
    }
}

File: samples/kotlin/sqrt/sqrt.kt
fun findSqrt(num: Double): Double {
    var x = num;
    var y: Double = 1.0;
    val e: Double = 0.000001;
    while (x-y > e) {
        x = (x + y) / 2;
        y = num / x;
    }
    return x;
}

fun main(args: Array<String>) {
    if (args.size != 1) {
        System.err.println("Error: Expected to receive an argument <number>");
        System.exit(1);
    }
    var number: Double = 0.0;
    try {
        number = args[0].toDouble();
    } catch (e: NumberFormatException) {
        System.err.println("Error: Expected to receive a number. Received '" + args[0] + "'");
        System.exit(1);
    }
    print(findSqrt(number).toString() + "\n");
}
